internal/staticanalysis/obfuscation/detection/generate_signals.go

```diff
@@ -0,0 +1,102 @@
+package detection
+
+import (
+	"fmt"
+	"github.com/ossf/package-analysis/internal/staticanalysis/obfuscation/parsing"
+	"github.com/ossf/package-analysis/internal/staticanalysis/obfuscation/stats"
+	"github.com/ossf/package-analysis/internal/staticanalysis/obfuscation/stringentropy"
+	"strings"
+)
+
+func getStrings(data *parsing.ParseResult) []string {
+	var extractedStrings []string
+	for _, d := range data.Literals {
+		switch d.Type {
+		case "string":
+			extractedStrings = append(extractedStrings, d.Value.(string))
+		case "float64":
+		case "bool":
+		}
+	}
+	return extractedStrings
+}
+
+func getIdentifierNames(data *parsing.ParseResult) []string {
+	var identifierNames []string
+	for _, ident := range data.Identifiers {
+		switch ident.Type {
+		case parsing.Function:
+			fallthrough
+		case parsing.Class:
+			fallthrough
+		case parsing.Parameter:
+			fallthrough
+		case parsing.Variable:
+			identifierNames = append(identifierNames, ident.Name)
+		}
+	}
+	return identifierNames
+}
+
+// characterAnalysis
+// Performs analysis on a collection of string symbols, returning:
+// - Stats summary of symbol (string) lengths
+// - Stats summary of symbol (string) entropies
+// - Entropy of all symbols concatenated together
+func characterAnalysis(symbols []string) (
+	lengthSummary stats.SampleStatistics,
+	entropySummary stats.SampleStatistics,
+	combinedEntropy float64,
+) {
+	// measure character probabilities by looking at entire set of strings
+	characterProbs := stringentropy.CharacterProbabilities(symbols)
+
+	var entropies []float64
+	var lengths []int
+	for _, s := range symbols {
+		entropies = append(entropies, stringentropy.CalculateEntropy(s, characterProbs))
+		lengths = append(lengths, len(s))
+	}
+
+	lengthSummary = stats.Summarise(lengths)
+	entropySummary = stats.Summarise(entropies)
+	combinedEntropy = stringentropy.CalculateEntropy(strings.Join(symbols, ""), nil)
+	return
```

Comment:
no need for explicit return

Reply:
I think it's necessary - if I delete it, I get a red underline saying "missing return statement"